x-pack/plugins/lists/server/services/exception_lists/utils/validate_data.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as t from 'io-ts';
+import { pipe } from 'fp-ts/pipeable';
+import { fold } from 'fp-ts/Either';
+import { exactCheck, formatErrors } from '@kbn/securitysolution-io-ts-utils';
+
+import { DataValidationError } from './errors';
+
+/**
+ * Validates that some data is valid by using an `io-ts` schema as the validator.
+ * Returns either `undefined` if the data is valid, or a `DataValidationError`, which includes
+ * a `reason` property with the errors encountered
+ * @param validator
+ * @param data
+ */
+export const validateData = <D>(validator: t.Type<D>, data: D): undefined | DataValidationError => {
```

Comment:
FYI: this utility is not very portable just yet. It seems to handle using `io-ts` schemas crated with `t.type()` or `t.intersection()`, but not others (like `t.exact()`). I did spent some time trying to see if I could find a way to alter the type here in order to support others, but was not successful. Since we don't need the other types (yet?), I'm deferring that change to the future.